src/Environments.jl

```diff
@@ -1,27 +1,174 @@
+
+    return basins_list
+end
+
+
 # Static Data Objects
-struct StaticEnvironment
-    "Directory containing list of all basins"
-    basins_dir::String
-    "Directory containing attribute csv"
-    attributes_dir::String
-    "Basin mapping dictionary"
+"""
+$(TYPEDEF)
+
+Stores the static features that describe the river basin network
+
+$(TYPEDFIELDS)
+"""
+struct StaticEnvironment{AV <: AbstractVector}
+    "Vector of basin identifiers"
+    basin_ids::AV
+    "Data frame of static basin attributes"
+    attributes::DataFrame
+    "Dictionary of pairs `(basin_id => basin_ids of direct upstream neighbours)`"
     graph_dict::Dict
 end
 
+"""
+$(TYPEDSIGNATURES)
+
+Constructor of `StaticEnvironment` from three strings holding files for basin id (txt), attributes (csv) and the graph (JSON).
+"""
+function StaticEnvironment(
+    basin_ids_file::AS1,
+    attributes_file::AS2,
+    graph_file::AS3,
+) where {AS1 <: AbstractString, AS2 <: AbstractString, AS3 <: AbstractString}
+
+    # create basin
+    basin_ids = get_basin_list(basin_ids_file)
+
+    # create attributes
+    attributes = CSV.read(attributes_file, DataFrame)
+
+    # create graph
+    graph_dict = JSON.parsefile(graph_file)
+
+    return StaticEnvironment(basin_ids, attributes, graph_dict)
+end
+
 # Dynamic Data Objects
+"""
+$(TYPEDEF)
+
+Stores the dynamic features that apply to the river network over a dated time period
+
+$(TYPEDFIELDS)
+"""
 struct DynamicEnvironment
-    "Directory containing forcing timeseries csv"
-    forcing_timeseries_dir::String
+    "Dictionary of pairs `(basin_id => forcing timeseries at basin_id)`"
+    forcing_timeseries::Dict
     "Directory to store simulation results"
     output_dir::String
 end
 
+"""
+$(TYPEDSIGNATURES)
+
+Constructor of `DynamicEnvironment` from a vector of `basin_id`s, and a vector of `forcing_timeseries_files` files (CSV).
+"""
+function DynamicEnvironment(
+    basin_ids::AV1,
+    forcing_timeseries_files::AV2,
+    output_dir::String,
+) where {AV1 <: AbstractVector, AV2 <: AbstractVector}
+
+    # build forcing timeseries
+    forcing_timeseries_array = []
+    for file in forcing_timeseries_files
+        push!(forcing_timeseries_array, CSV.read(file, DataFrame))
+    end
+    forcing_timeseries = Dict(eachrow([basin_ids forcing_timeseries_array])) # creates id => timeseries dictionary
+
+    return DynamicEnvironment(forcing_timeseries, output_dir)
+end
+
+
+"""
+$(TYPEDEF)
+
+Stores both the static and dynamic environments
+
+$(TYPEDFIELDS)
+"""
 struct Environment{SE <: StaticEnvironment, DE <: DynamicEnvironment}
-    "Static data objects"
+    "StaticEnvironment data objects"
     static_env::SE
-    "Dynamic data objects"
+    "DynamicEnvironment data objects"
     dynamic_env::DE
 end
+
+"""
+$(TYPEDSIGNATURES)
+
+Constructor of `Enviroment` using a list of forcing timeseries files. See constructors for StaticEnvironment and DynamicEnvironment for more details on other inputs.
+"""
+function Environment(
```

Comment:
It might be nice to make these kwargs since there are several of them of the same type and the current function input would require knowing the order

Reply:
This is the main reason why I made the docstring. But happy to make a keyword version!